jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/listener/JMSExceptionListener.java

```diff
@@ -0,0 +1,50 @@
+package org.wso2.carbon.transport.jms.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.transport.jms.exception.JMSConnectorException;
+
+import javax.jms.ExceptionListener;
+import javax.jms.JMSException;
+
+/**
+ * Listener that listens to the problem in the jms connection.
+ */
+class JMSExceptionListener implements ExceptionListener {
```

Comment:
We register this listener, during the connection creation time. As mentioned in the java-doc http://docs.oracle.com/javaee/6/api/javax/jms/ExceptionListener.html, this will only be triggered if the connection problem cannot be handled by the JMS provider.